EventsExpress/ClientApp/src/actions/login.js

```diff
@@ -3,71 +3,89 @@ import { func } from 'prop-types';
 import { initialConnection } from './chat';
 import { getUnreadMessages } from './chats';
 
-
 export const SET_LOGIN_PENDING = "SET_LOGIN_PENDING";
 export const SET_LOGIN_SUCCESS = "SET_LOGIN_SUCCESS";
 export const SET_LOGIN_ERROR = "SET_LOGIN_ERROR";
 export const SET_USER = "SET_USER";
 
-
-
 const api_serv = new EventsExpressService();
 
 export default function login(email, password) {
-
   return dispatch => {
     dispatch(setLoginPending(true));
 
-    const res = api_serv.setLogin({Email: email, Password: password});
-    res.then(response => {
-      if(response.error == null){
-          dispatch(setUser(response));
-          dispatch(setLoginSuccess(true));
-         
-          localStorage.setItem('token', response.token);
-          
-          localStorage.setItem('id', response.id);
-          dispatch(initialConnection());
+    const res = api_serv.setLogin({ Email: email, Password: password });
 
-          dispatch(getUnreadMessages(response.id));
-        }else{
-          dispatch(setLoginError(response.error));
-        }
-      });
+    res.then(response => {
+      if (response.error == null) {
+        dispatch(setUser(response));
+        dispatch(setLoginSuccess(true));
+        localStorage.setItem('token', response.token);
+        localStorage.setItem('id', response.id);
+        dispatch(initialConnection());
+        dispatch(getUnreadMessages(response.id));
+      } else {
+        dispatch(setLoginError(response.error));
+      }
+    });
   }
 }
-export function loginFacebook(email, name) {
 
-    return dispatch => {
-        dispatch(setLoginPending(true));
+export function loginGoogle(tokenId, email, name, imageUrl) {
+  return dispatch => {
+    dispatch(setLoginPending(true));
 
-        const res = api_serv.setFacebookLogin({ Email: email, Name: name });
-        res.then(response => {
-            if (response.error == null) {
-                dispatch(setUser(response));
-                dispatch(setLoginSuccess(true));
-                localStorage.setItem('token', response.token);
-                localStorage.setItem('id', response.id);
-                dispatch(initialConnection());
+    const res = api_serv.setGooglekLogin({
+      TokenId: tokenId,
+      Email: email,
+      Name: name,
+      PhotoUrl: imageUrl
+    });
 
-                dispatch(getUnreadMessages(response.id));
-            } else {
-                dispatch(setLoginError(response.error));
-            }
-        });
-    }
+    res.then(response => {
```

Comment:
This code is complete duplication of the code for `login` function. Could you please extract it into a private method and call it in two places.